openmc/filter.py

```diff
@@ -132,18 +135,18 @@ def bins(self, bins):
         if self._type in ['cell', 'cellborn', 'surface', 'material',
                           'universe', 'distribcell']:
             for edge in bins:
-                if not is_integer(edge):
+                if not isinstance(edge, Integral):
                     msg = 'Unable to add bin "{0}" to a {1} Filter since ' \
-                          'it is a non-integer'.format(edge, self._type)
+                          'it is not an integer'.format(edge, self._type)
                     raise ValueError(msg)
                 elif edge < 0:
                     msg = 'Unable to add bin "{0}" to a {1} Filter since ' \
-                          'it is a negative integer'.format(edge, self._type)
+                          'it is negative'.format(edge, self._type)
                     raise ValueError(msg)
 
         elif self._type in ['energy', 'energyout']:
             for edge in bins:
-                if not is_integer(edge) and not is_float(edge):
+                if not isinstance(edge, Real):
```

Comment:
It could, but the default error message wouldn't make sense, i.e. "can't set __ to ___".